src/Microsoft.IdentityModel.Validators/AadValidationParametersExtension.Internal.cs

```diff
@@ -0,0 +1,254 @@
+
+                ValidateSigningKeyCloudInstance(securityKey, configuration);
+
+                // preserve and run provided logic
+                if (issuerSigningKeyValidationDelegate != null)
+                    return issuerSigningKeyValidationDelegate(securityKey, securityToken, validationParameters, callContext);
+
+                return new ValidatedSigningKeyLifetime(DateTime.UtcNow, DateTime.UtcNow, DateTime.UtcNow);
+            };
+
+            validationParameters.IssuerSigningKeyValidator = issuerSigningKeyValidationDelegate1;
+        }
+
+        /// <summary>
+        /// Enables the validation of the issuer of the signing keys used by the Microsoft identity platform (AAD) against the issuer of the token.
+        /// </summary>
+        /// <param name="validationParameters">The <see cref="TokenValidationParameters"/> that are used to validate the token.</param>
+        internal static void EnableAadSigningKeyIssuerValidation(this ValidationParameters validationParameters)
+        {
+            if (validationParameters == null)
+                throw LogHelper.LogArgumentNullException(nameof(validationParameters));
+
+            IssuerSigningKeyValidationDelegate issuerSigningKeyValidationDelegate = validationParameters.IssuerSigningKeyValidator;
+
+            validationParameters.IssuerSigningKeyValidator = (securityKey, securityToken, vp, callContext) =>
+            {
+                BaseConfiguration baseConfiguration = null;
+                if (vp.ConfigurationManager != null)
+                    baseConfiguration = vp.ConfigurationManager.GetBaseConfigurationAsync(System.Threading.CancellationToken.None).ConfigureAwait(false).GetAwaiter().GetResult();
+
+                ValidateIssuerSigningKey(securityKey, securityToken, baseConfiguration);
+
+                // preserve and run provided logic
+                if (issuerSigningKeyValidationDelegate != null)
+                    return issuerSigningKeyValidationDelegate(securityKey, securityToken, vp, callContext);
+
+                return ValidateIssuerSigningKeyCertificate(securityKey, validationParameters);
+            };
+        }
+
+        /// <summary>
+        /// Validates the issuer signing key.
+        /// </summary>
+        /// <param name="securityKey">The <see cref="SecurityKey"/> that signed the <see cref="SecurityToken"/>.</param>
+        /// <param name="securityToken">The <see cref="SecurityToken"/> being validated, could be a JwtSecurityToken or JsonWebToken.</param>
+        /// <param name="configuration">The <see cref="BaseConfiguration"/> provided.</param>
+        /// <returns><c>true</c> if the issuer of the signing key is valid; otherwise, <c>false</c>.</returns>
+        internal static bool ValidateIssuerSigningKey(SecurityKey securityKey, SecurityToken securityToken, BaseConfiguration configuration)
```

Comment:
This method is common with AadTokenValidationParametersExtensions (to the letter)